portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -60,16 +73,23 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
    * Iff request has a non-null comment-text parameter,
    *   then create and store the parameter's String as a Datastore Comment Object
    * Always redirect to index.html
+   * If poster not logged in, return a HTTP 400 error code
    */
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    String comment = request.getParameter("comment-text");
-    if(comment != null){
-      long timestamp = System.currentTimeMillis();
-      Entity commentEn = new Entity("Comment");
-      commentEn.setProperty("text",comment);
-      commentEn.setProperty("timestamp",timestamp);
-      datastore.put(commentEn);
+    if(userService.isUserLoggedIn()) {
+      String comment = request.getParameter("comment-text");
+      if (comment != null) {
+        String authorId = userService.getCurrentUser().getUserId();
+        long timestampMs = System.currentTimeMillis();
+        Entity commentEn = new Entity("Comment");
+        commentEn.setProperty("comment-text", comment);
+        commentEn.setProperty("timestamp-ms", timestampMs);
+        commentEn.setProperty("author-id", authorId);
+        datastore.put(commentEn);
+      }
+    } else {
+      response.setStatus(400);
```

Comment:
Perhaps an error 403?  Would that be better?

Reply:
I was thinking that if you aren't logged in you're by default making a bad request, but since its caused by not logging on I think 403 is more descriptive. I'll change it to that.